src/AWS.Deploy.Common/Recipes/Validation/FargateTaskSizeCpuMemoryLimitsRecipeValidator.cs

```diff
@@ -0,0 +1,92 @@
+    /// <para/> 2048 (2 vCPU)   Between 4096 (4 GB) and 16384 (16 GB) in increments of 1024 (1 GB) 
+    /// <para/> 4096 (4 vCPU)   Between 8192 (8 GB) and 30720 (30 GB) in increments of 1024 (1 GB)
+    /// <para />
+    /// See https://docs.aws.amazon.com/AmazonECS/latest/userguide/task_definition_parameters.html#task_size
+    /// for more details.
+    /// </summary>
+    public class FargateTaskSizeCpuMemoryLimitsRecipeValidator : IRecipeValidator
+    {
+        private readonly Dictionary<string, string[]> _cpuMemoryMap = new()
+        {
+            { "256", new[] { "512", "1024", "2048" } },
+            { "512", new[] { "1024", "2048", "3072", "4096" } },
+            { "1024", new[] { "2048", "3072", "4096", "5120", "6144", "7168", "8192" } },
+            { "2048", BuildMemoryArray(4096, 16384).ToArray() },
+            { "4096", BuildMemoryArray(8192, 30720).ToArray()}
+        };
+
+        private static IEnumerable<string> BuildMemoryArray(int start, int end, int increment = 1024)
+        {
+            while (start <= end)
+            {
+                yield return start.ToString();
+                start += increment;
+            }
+        }
+
+        /// <summary>
+        /// Supports replacement tokens {{cpu}}, {{memory}}, and {{memoryList}}
+        /// </summary>
+        public string ValidationFailedMessage { get; set; } =
+            "Cpu value {{cpu}} is not compatible with value {{memory}}.  Allowed values are {{memoryList}}";
+
+        public string? InvalidCpuValueValidationFailedMessage {get;set;}
+
+        /// <inheritdoc cref="FargateTaskSizeCpuMemoryLimitsRecipeValidator"/>
+        public ValidationResult Validate(RecipeDefinition recipe, IDeployToolValidationContext deployValidationContext)
+        {
+            var cpuItem = recipe.OptionSettings.FirstOrDefault(x => x.Id == "TaskCpu");
+            var memoryItem = recipe.OptionSettings.FirstOrDefault(x => x.Id == "TaskMemory");
+
+            // If we can't find the cpu/memory OptionSettingItem
+            // bail out as 'valid'
+            if (null == cpuItem || null == memoryItem)
+            {
+                return ValidationResult.Valid();
+            }
+
+            var cpu = cpuItem.GetValue<string>(new Dictionary<string, string>());
+            var memory = memoryItem.GetValue<string>(new Dictionary<string, string>());
+
+            if (!_cpuMemoryMap.ContainsKey(cpu))
+            {
+                // this could happen, but shouldn't.
+                // either there is mismatch between _cpuMemoryMap and the AllowedValues
+                // or the UX flow calling in here doesn't enforce AllowedValues.
+                var message = InvalidCpuValueValidationFailedMessage?.Replace("{{cpu}}", cpu);
+
+                return ValidationResult.Failed(message?? "Cpu validation failed");
+            }
+
+            var validMemoryValues = _cpuMemoryMap[cpu];
+
+            if (validMemoryValues.Contains(memory))
```

Comment:
nit: the list of valid values is now small and i dont know how big it could get, but should we consider using a dictionary instead of an array?

Reply:
Changing it to a dictionary is not required because even if this list got as big as having 100 elements, the `List.Contains(..)` operation will still be O(100) which is ~ O(1).